app/main.py

```diff
@@ -1,3 +1,63 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.Shop import Shop
+from app.Car import Car
+from app.Customer import Customer
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        data = json.load(file)
+
+    fuel_price = data["FUEL_PRICE"]
+    customers_data = data["customers"]
+    shops_data = data["shops"]
+
+    shops = []
+    for shop_info in shops_data:
+        shop_name = shop_info["name"]
+        shop_location = shop_info["location"]
+        products = shop_info["products"]
+        shop = Shop(shop_name, shop_location, products)
+        shops.append(shop)
+
+    for customer_data in customers_data:
+        customer_car = Car(
+            customer_data["car"]["brand"],
+            customer_data["car"]["fuel_consumption"])
+        customer = Customer(
+            customer_data["name"],
+            customer_data["product_cart"],
+            customer_data["location"],
+            customer_data["money"],
+            customer_car)
+
+        min_cost = float("inf")
+        cheap_shop = None
+
+        customer.customer_money()
+        for shop in shops:
+            total_cost = customer.distance(
+                shop.shop_location, customer_car.fuel_consumption, fuel_price)
+            total_cost += sum(
+                price * shop.products[product_name]
+                for product_name, price in customer.product_cart.items())
```

Comment:
In case you leave a bracket on a line without arguments, your closing bracket should not have arguments on the same line as well.

Reply:
still not fixed. closing parenthesis should be on new line